### PR TITLE
AutoPan effect improvements

### PR DESCRIPTION
**Pull request by Be-ing**
* Use a narrower range for the period parameter
  * Link period and smoothing parameters to the metaknob, with period inversely linked
  * Narrow the range of the smoothing parameter because not using any smoothing sounds bad.
  * Improve parameter descriptions

Depends on:
  - [x] #1256 (for roundToFraction function)

### CODE REVIEW

**Comment by daschuer** [comment]
I think there is an underlying issue how smoothing is implemented. I have not tested it recently. Am I right that smoothing 0 goes instantly from the right to middle to left? I think the middle step is the one which feels unnatural, because it does not match a de:"Schunkeln" move.
I think smoothing 0 schould go directly  from hard left to hard right.

**Comment by Be-ing** [comment]
When I listen in headphones with smoothing at 0, the sound goes jumps entirely from one side to the other with nothing in between, which sounds weird.

File: src/effects/native/autopaneffect.cpp [hunk 1]
```diff
@@ -114,31 +115,13 @@ void AutoPanEffect::processChannel(const ChannelHandle& handle, PanGroupState* p
...
-        }
-
-        double beats = m_pPeriodParameter->minimum();
-        while (i != 0.0) {
-            beats *= 2;
-            i--;
-        }
-
+        // period is a number of beats
+        double beats = std::max(roundToFraction(period, 2), 0.25);
```
Comment (daschuer) [inline src/effects/native/autopaneffect.cpp]: error: ‘roundToFraction’ was not declared in this scope

Comment (Be-ing) [inline src/effects/native/autopaneffect.cpp]: It's added in #1256.

Comment (daschuer) [inline src/effects/native/autopaneffect.cpp]: Oh sorry, you have stated this already.

**Comment by daschuer** [comment]
> Narrow the range of the smoothing parameter because not using any smoothing sounds bad.

This sounds bad indeed on headphones, but not on a real PA system. This could be a wanted effect. 
Please read 0 as minimum. Could you also normalize smoothing to 1? This is part of the external interface and no one will understand 0.5 as maximum without explaining.

**Comment by daschuer** [comment]
There is something completely wrong with the beat syncing code. 
If I put it to the most famous German schunkel song: https://www.youtube.com/watch?v=wYus-2DGCtM :-P
(Mix fails to get the right beat, you need to 2/3 it to 62.83)
On 1 Beat, I expect to schunkel like these guys with the sound of the speakers. 
But the it shifts slightly off beat. If you can stand to listen to the middle of the track, the effect is complete of beat. 

Does it make sense to sync to 2 1/2 beats? Probably not. It is quite hard to set the desired beat count without developer tool tips enabled.

The rate slider is not considered, I think this should be. I am not sure if we should consider scratching as well?

**Comment by Be-ing** [comment]
> Does it make sense to sync to 2 1/2 beats? Probably not. It is quite hard to set the desired beat count without developer tool tips enabled.

Not really, but does it make sense for the knob to have a big gap between 2 and 3 beats where it does nothing?

**Comment by mevsme** [comment]
AutoPan is a great feature in general. Also it helped me to check if my L\R VU meters works fine.
But I have a feedback.

Here's how it "feels" for SYNC ON in Deere


I think I should be working like music producers used to effects are workin in SyncMode, e.g.: 
I tried two DAWs.
1) in FL studio it's a bit weird to me, cuz I can not guess what numbers 1:00 2:00 3:45 means. If it synced or not.
2) In Cubase it works like this:
 1/2, 1/4, 1/8, 1/16/ 1/32 (32-very fast and yes, from slower to faster)
than 1/1 T, 1/2 T, 1/4 T (triplets)
than 1/1 D, 1/2 D (in 1/2D feedback is on 3rd quater in 4/4 bar)

if you turn off sync it has freqs from 0.1ms to 5000ms. 
I think it is right to modify the rotary into three equal parts and adjust speed from slower to faster not very fast to very very slooooow in linear mode. 



Is it possible at all or this is weird request only I need? :laughing: 
...but yeah, that is not so "obvious" how it works when you are using a controller rotary...

At least maybe we should make a bigger ramp for "quick values" e.g. speeds that are between 0-50% becomes 0-75%. Does somebody understand what I am saying? :smile:

**Comment by Be-ing** [comment]
Triplets are cool, but I don't really like the idea of different zones that behave differently across the range of one parameter. I don't know what the best way to enable the use of triplets is. Maybe a binary button parameter? Anyway, I think that is beyond the scope of this PR.

The period range was previously 1/16 beat to 128 beats on a logarithmic scale, which I agree is nonsensical. I have narrowed it to 1/4 to 4 beats on a linear scale here.

**Comment by mevsme** [comment]
Yeah, maybe you are right, it is impossible to implement such a GUI to make it clear what is what in different zones of a rotary.

I have another question, can some rotary be added to the Autopan to make it manual-pan :sweat_smile: ? So you could pan sound by rotating super FX super knob?

**Comment by Be-ing** [comment]
No, that is a different effect that is being worked on in #1294.

**Comment by daschuer** [comment]
Ups, I pushed to the wrong branch. Reverted.

**Comment by Be-ing** [comment]
I have removed the Sync parameter. Like Phaser and Flanger, I do not think there is a use case for an unsynced AutoPan. I tried adding a Triplet parameter, but unlike Phaser, Flanger, and Echo, I do not think there is a use case for that with this effect.

**Comment by daschuer** [comment]
Thank you! LGTM

**Pull request merged**